src/platform/plugins/shared/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -619,6 +620,8 @@ export class SearchInterceptor {
             ) {
               this.showRestoreWarning(sessionId);
             }
+
+            deepFreeze(response);
```

Comment:
@thomasneirynck correct me if I'm wrong here– I think we'd like to take this out of this specific PR. It does help surface issues as they come up but it may also cause issues to arise during runtime that we don't have specific unit or functional tests for. We also believe this may be the main cause of the performance regression. I've opened https://github.com/elastic/kibana/issues/223960 to make an attempt at fixing this through types (I know this won't catch everything but it will at least help).